crates/task-impls/src/helpers.rs

```diff
@@ -44,7 +44,7 @@ pub async fn broadcast_event<E: Clone + std::fmt::Debug>(event: E, sender: &Send
 /// Panics if the VID calculation fails, this should not happen.
 #[allow(clippy::panic)]
 pub async fn calculate_vid_disperse<TYPES: NodeType>(
-    txns: Vec<u8>,
+    txns: Arc<[u8]>,
```

Comment:
same as above:
> Yes, this is tricky with lifetimes because txns is passed into spawn_blocking. Maybe there is another more elegant approach but keeping this as an Arc<[u8]> and just handing out a reference seems to work.